homeassistant/components/egardia.py

```diff
@@ -0,0 +1,139 @@
+        vol.Optional(CONF_REPORT_SERVER_PORT,
+                     default=DEFAULT_REPORT_SERVER_PORT): cv.port,
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    """Set up the Egardia platform."""
+    from pythonegardia import egardiadevice
+    from pythonegardia import egardiaserver
+
+    name = config[DOMAIN].get(CONF_NAME)
+    hass.data[D_EGARDIANM] = name
+    username = config[DOMAIN].get(CONF_USERNAME)
+    password = config[DOMAIN].get(CONF_PASSWORD)
+    host = config[DOMAIN].get(CONF_HOST)
+    port = config[DOMAIN].get(CONF_PORT)
+    rs_enabled = config[DOMAIN].get(CONF_REPORT_SERVER_ENABLED)
+    hass.data[D_EGARDIARSENABLED] = rs_enabled
+    rs_port = config[DOMAIN].get(CONF_REPORT_SERVER_PORT)
+    rs_codes = config[DOMAIN].get(CONF_REPORT_SERVER_CODES)
+    hass.data[D_EGARDIARSCODES] = rs_codes
+    version = config[DOMAIN].get(CONF_VERSION)
+    try:
+        hass.data[D_EGARDIASYS] = egardiadevice.EgardiaDevice(
+            host, port, username, password, '', version)
+    except requests.exceptions.RequestException:
+        raise exc.PlatformNotReady()
+    except egardiadevice.UnauthorizedError:
+        _LOGGER.error("Unable to authorize. Wrong password or username")
+        return
+
+    # add egardia alarm device
+    def alarm_loaded(event, data=None):
+        """Check if egardia platform has loaded."""
+        if event is DOMAIN:
+            # configure egardia server, including callback
+            if rs_enabled:
+                # Set up the egardia server
+                _LOGGER.info("Setting up EgardiaServer")
+                try:
+                    if D_EGARDIASRV not in hass.data:
+                        server = egardiaserver.EgardiaServer('', rs_port)
+                        bound = server.bind()
+                        if not bound:
+                            raise IOError("Binding error occurred while " +
+                                          "starting EgardiaServer")
+                        hass.data[D_EGARDIASRV] = server
+                        server.start()
+                except IOError:
+                    return
+                hass.data[D_EGARDIASRV].register_callback(
+                    hass.data[D_EGARDIADEV].handle_status_event)
+
+    # register for callback since we might need to set up the egardia server
+    discovery.listen_platform(hass, 'alarm_control_panel', alarm_loaded)
+
+    hass.async_add_job(discovery.async_load_platform(
+        hass, 'alarm_control_panel', DOMAIN,
+        discovered=None, hass_config=config))
+
+    def handle_stop_event(event):
+        """Callback function for HA stop event."""
+        hass.data[D_EGARDIASRV].stop()
+
+    # listen to home assistant stop event
+    hass.bus.listen_once(EVENT_HOMEASSISTANT_STOP, handle_stop_event)
+
+    # get the sensors from the device and add those
+    sensors = hass.data[D_EGARDIASYS].getsensors()
+    hass.async_add_job(discovery.async_load_platform(
```

Comment:
Call the sync version `load_platform` directly. Don't add a job to call the async version.

Reply:
Done.